caravel/assets/javascripts/explorev2/components/ChartControl.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import Select from 'react-select';
+import { bindActionCreators } from 'redux';
+import * as actions from '../actions/exploreActions';
+import { connect } from 'react-redux';
+import { VIZ_TYPES } from '../../../utils/common';
+
+class ChartControl extends React.Component {
+  componentWillMount() {
+    if (this.props.datasourceId) {
+      this.props.actions.setFormOpts(this.props.datasourceId, this.props.datasourceType);
+    }
+  }
+  changeDatasource(datasourceOpt) {
+    const val = (datasourceOpt) ? datasourceOpt.value : null;
+    this.props.actions.setDatasource(val);
+    this.props.actions.resetFormData();
+    this.props.actions.setFormOpts(val, this.props.datasourceType);
+  }
+  changeViz(vizOpt) {
+    const val = (vizOpt) ? vizOpt.value : null;
+    this.props.actions.setVizType(val);
+  }
+  render() {
+    return (
+      <div className="panel space-1">
+        <div className="panel-header">Chart Options</div>
+        <div className="panel-body">
+          <h5 className="section-heading">Datasource</h5>
+          <div className="row">
+            <Select
+              name="select-datasource"
+              placeholder={'Select a datasource'}
+              options={this.props.datasources.map((d) => ({ value: d[0], label: d[1] }))}
+              value={this.props.datasourceId}
+              autosize={false}
+              onChange={this.changeDatasource.bind(this)}
+            />
+          </div>
+          <h5 className="section-heading">Viz Type</h5>
+          <div className="row">
+            <Select
+              name="select-viztype"
+              placeholder={'Select a viz type'}
+              options={VIZ_TYPES}
+              value={this.props.vizType}
+              autosize={false}
+              onChange={this.changeViz.bind(this)}
+            />
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+ChartControl.propTypes = {
+  actions: React.PropTypes.object,
```

Comment:
I saw Sqllab components were doing this, I think it's for mapDispatchToProps, so that we can bind action creators into one, then dispatch from the component props